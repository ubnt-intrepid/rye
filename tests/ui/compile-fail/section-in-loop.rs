fn main() {}

#[rye::test]
fn section_in_loop() {
    loop {
        section!("section", {
            assert!(true);
        });
    }
}

#[rye::test]
fn section_in_for_loop() {
    for _ in 0..10 {
        section!("section", {
            assert!(true);
        });
    }
}

#[rye::test]
fn section_in_while_loop() {
    while false {
        section!("section", {
            assert!(true);
        });
    }
}
