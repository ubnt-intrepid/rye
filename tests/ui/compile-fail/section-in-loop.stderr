error: section cannot be described in a loop
 --> $DIR/section-in-loop.rs:6:9
  |
6 | /         section!("section", {
7 | |             assert!(true);
8 | |         });
  | |__________^

error: section cannot be described in a loop
  --> $DIR/section-in-loop.rs:15:9
   |
15 | /         section!("section", {
16 | |             assert!(true);
17 | |         });
   | |__________^

error: section cannot be described in a loop
  --> $DIR/section-in-loop.rs:24:9
   |
24 | /         section!("section", {
25 | |             assert!(true);
26 | |         });
   | |__________^
